app/controllers/projects_controller.rb

```diff
@@ -21,6 +25,15 @@ def sort
     head :ok
   end
 
+  def toggle_archive
+    @project = Project.find(params[:project_id])
+    if !@project.archived?
+      @project.update_column :status, "archived"
+    else
+      @project.update_column :status, nil
+    end
```

Comment:
What do you thinkg about something like:

```rb
def toggle_archived
  Project.find(params[:id]).toggle_archived! # check this should be params[:id], not params[:project_id]
end
```

and then in the model something like:

```rb
def toggle_archived!
  new_status = archived? ? nil : "archived"
  update_column :status, new_status
end
```